### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by davidrapan** [davidrapan/ha-solarman#78] [Announcements]
I added wiki page which contains modbus documents I have and used: [Documentation](https://github.com/davidrapan/ha-solarman/wiki/Documentation)

**If any of you would like to add some other devices use this thread!**

**Comment by batkite** [comment]
Hi,

I've created a yaml for my inverter : KStar BlugG 3000D Single-Phase if you want to add it to the list (it's in French but I'll also give you a tab with the various registers ):

### KSTAR BlueG Modbus Register Map (Input Registers - Function Code 04)

| Register | Parameter Name | Unit | Scale | Description |
| :--- | :--- | :--- | :--- | :--- |
| **3000** | PV1 Voltage | V | 0.1 | DC voltage from solar string 1 |
| **3001** | PV2 Voltage | V | 0.1 | DC voltage from solar string 2 |
| **3003** | PV1 Current | A | 0.01 | DC current from solar string 1 |
| **3004** | PV2 Current | A | 0.01 | DC current from solar string 2 |
| **3007** | PV1 Power | W | 1 | Raw DC power input from string 1 |
| **3009** | PV2 Power | W | 1 | Raw DC power input from string 2 |
| **3014** | Grid Voltage | V | 0.1 | AC grid voltage |
| **3017** | Grid Frequency | Hz | 0.01 | AC grid frequency |
| **3020** | Grid Current | A | 0.01 | AC current injected into the grid |
| **3024** | Current AC Power | W | 1 | Real-time AC power output |
| **3025** | Inverter Temperature | °C | 0.1 | Internal temperature of the inverter |
| **3039** | Total Production | kWh | 0.1 | Cumulative yield since installation |
| **3041** | Yearly Production | kWh | 1 | Cumulative yield for the current year |
| **3097** | Peak Power History | W | 1 | Historical highest peak recorded |
| **3099** | Peak Power Today | W | 1 | Highest peak recorded today |
| **3101** | Weekly Production | kWh | 1 | Cumulative yield for the current week |
| **3103** | Monthly Production | kWh | 1 | Cumulative yield for the current month |
| **3211** | Today Production | kWh | 0.1 | Daily yield (reset at midnight) |

[KSTAR-BlueG-3000D.yaml](https://github.com/user-attachments/files/26662600/KSTAR-BlueG-3000D.yaml)
